pkg/controller/metadata/metadata_controller.go

```diff
@@ -348,14 +363,21 @@ func (r *ReconcileMetadata) Reconcile(request reconcile.Request) (reconcile.Resu
 		}
 	} else if err != nil {
 		return reconcile.Result{}, err
-	} else {
-		// TODO: Update deployment if changed
+	} else if foundDeployment.ObjectMeta.Annotations[versionAnnotation] != currentVersion {
+		log.Info("Updating Deployment", "namespace", metadataDeployment.Namespace, "name", metadataDeployment.Name)
+		err = r.Update(context.TODO(), foundDeployment)
+		if err != nil {
+			return reconcile.Result{}, err
```

Comment:
Should this be wrapped with a "Failed to update deployment" as per Secret above?

Reply:
👍